scripts/benchmark/utils/chain_plumbing.py

```diff
@@ -81,19 +92,28 @@
 GenesisState = Iterable[Tuple[Address, Dict[str, Any]]]
 
 
-def get_chain(vm: Type[BaseVM], genesis_state: GenesisState) -> MiningChain:
+def get_chain(vm: Type[BaseVM], chain_db: Type[BaseDB], genesis_state: GenesisState) -> MiningChain:
 
     chain = build(
         MiningChain,
         fork_at(vm, constants.GENESIS_BLOCK_NUMBER),
         disable_pow_check(),
-        genesis(params=GENESIS_PARAMS, state=genesis_state)
+        genesis(db=chain_db, params=GENESIS_PARAMS, state=genesis_state)
     )
 
     return chain
 
 
 def get_all_chains(genesis_state: GenesisState=DEFAULT_GENESIS_STATE) -> Iterable[MiningChain]:
     for vm in ALL_VM:
-        chain = get_chain(vm, genesis_state)
-        yield chain
+        # Chain using MemoryDB
+        memory_db_obj = AtomicDB()
+        memory_db_chain = get_chain(vm, memory_db_obj, genesis_state)
+        yield memory_db_chain
+
+        # Chain using LevelDB
+        with tempfile.TemporaryDirectory() as temp_dir:
+            level_db_obj = LevelDB(temp_dir)
+            level_db_chain = get_chain(vm, level_db_obj, genesis_state)
+            yield level_db_chain
+        del temp_dir
```

Comment:
I think this is not necessary since you are using the context manager API to get the temporary directory.

Reply:
You mean the `del temp_dir` part right?